### PR TITLE
Move all the color calculation to a helper file

### PR DESCRIPTION
**Pull request by bembelimen**

### CODE REVIEW

File: administrator/templates/atum/Service/HTML/Atum.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage     com_content
```
Comment (brianteeman) [inline administrator/templates/atum/Service/HTML/Atum.php]: be careful when you copy paste

**Comment by brianteeman** [comment]
Surely this should all be in the css - isnt that what the _variables.css is for?

File: administrator/templates/atum/error_full.php [hunk 2]
```diff
@@ -92,18 +166,18 @@
...
+					<img src="<?php echo $siteLogo; ?>" alt="">
+					<img class="logo-small" src="<?php echo $smallLogo; ?>" alt="">
+				</a>
 			</div>
 			<jdoc:include type="modules" name="title"/>
 		</div>
-		<div class="header-items d-flex ml-auto">
-			<jdoc:include type="modules" name="status" style="header-element"/>
+		<div class="header-items d-flex">
+			<jdoc:include type="modules" name="status" style="header-item"/>
```
Comment (brianteeman) [inline administrator/templates/atum/error_full.php]: Not sure if we have anything in the codestyle docs on this but everywhere else it seems to be 
`<jdoc:include ........" />`

Comment (bembelimen) [reply administrator/templates/atum/error_full.php]: I created an issue: https://github.com/joomla/backend-template/issues/327

**Comment by bembelimen** [comment]
> 
> 
> Surely this should all be in the css - isnt that what the _variables.css is for?

No, this is the "calculation" of the params values you can set in the template style, that has nothing to do with SCSS, it's pue PHP.

**Comment by brianteeman** [comment]
got you - thanks for the explanation

**Comment by brianteeman** [comment]
I wont pretend that I am able to understand this code but I can confirm that everything works the same as it did before

File: administrator/templates/atum/Service/HTML/Atum.php [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	 *
+	 * @since  4.0.0
+	 */
+	public static function rootcolors(Registry $params) : void
+	{
+		$monochrome = (bool) $params->get('monochrome');
+
+		$root = [];
+		if ($params->exists('hue'))
+		{
```
Comment (Quy) [inline administrator/templates/atum/Service/HTML/Atum.php]:
```suggestion
		{
```

File: administrator/templates/atum/Service/HTML/Atum.php [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	 *
+	 * @return void
+	 *
+	 * @since  4.0.0
+	 */
+	public static function rootcolors(Registry $params) : void
+	{
+		$monochrome = (bool) $params->get('monochrome');
+
+		$root = [];
```
Comment (Quy) [inline administrator/templates/atum/Service/HTML/Atum.php]:
```suggestion
		$root = [];

```

File: administrator/language/en-GB/en-GB.ini [hunk 5]
```diff
@@ -196,6 +196,12 @@ JFIELD_BASIS_LOGOUT_DESCRIPTION_LABEL="Logout Description Text"
 JFIELD_BASIS_LOGOUT_DESCRIPTION_SHOW_DESC="Show or hide logout description."
 JFIELD_BASIS_LOGOUT_DESCRIPTION_SHOW_LABEL="Logout Description"
 JFIELD_CATEGORY_DESC="The category that this item is assigned to. You may select an existing category or enter a new category by typing the name in the field and pressing enter."
+JFIELD_COLOR_WRONG_FORMAT="Wrong format"
+JFIELD_COLOR_LABEL_SLIDER_INPUT="Selected Color Value"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.ini]:
```suggestion
JFIELD_COLOR_LABEL_SLIDER_INPUT="Selected Colour Value"
```

File: administrator/templates/atum/Service/HTML/Atum.php [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+use Joomla\CMS\Factory;
+use OzdemirBurak\Iris\Color\Hex;
+use OzdemirBurak\Iris\Color\Hsl;
+use OzdemirBurak\Iris\Color\Rgb;
+use OzdemirBurak\Iris\Color\Hsv;
+use OzdemirBurak\Iris\Color\Hsla;
+use Joomla\Registry\Registry;
+
+/**
+ * Content Component HTML Helper
```
Comment (brianteeman) [inline administrator/templates/atum/Service/HTML/Atum.php]: ????

File: administrator/templates/atum/templateDetails.xml [hunk 7]
```diff
@@ -50,13 +51,14 @@
...
 			<fieldset name="colors" label="TPL_ATUM_COLORS_SETTINGS_LABEL">
 				<field
-					name="hue"
-					type="color"
-					label="TPL_ATUM_COLORS_HUE"
-					format="hue"
-					control="slider"
-					preview="true"
-					default="#003e7c"
+						name="hue"
```
Comment (brianteeman) [inline administrator/templates/atum/templateDetails.xml:54]: Please re-order to that it starts Name, Type, Label

File: layouts/joomla/form/field/color/slider.php [hunk 8]
```diff
@@ -49,67 +57,89 @@
...
 $displayValues = explode(',', $display);
 $allSliders    = $display === 'full' || empty($display);
-$displayData   = ' data-display="' . ($display ? $display : 'full') . '"';
 
 HTMLHelper::_('stylesheet', 'system/fields/joomla-field-color-slider.min.css', ['version' => 'auto', 'relative' => true]);
 HTMLHelper::_('script', 'system/fields/joomla-field-color-slider.min.js', ['version' => 'auto', 'relative' => true]);
+
+Text::script('JFIELD_COLOR_ERROR_CONVERT_HSL');
+Text::script('JFIELD_COLOR_ERROR_CONVERT_HUE');
+Text::script('JFIELD_COLOR_ERROR_NO_COLOUR');
```
Comment (brianteeman) [inline layouts/joomla/form/field/color/slider.php:72]: Much as I hate to see the word spelled as color lets not confuse thing by using both ;)
```suggestion
Text::script('JFIELD_COLOR_ERROR_NO_COLOR');
```

File: administrator/language/en-GB/en-GB.ini [hunk 9]
```diff
@@ -196,6 +196,15 @@ JFIELD_BASIS_LOGOUT_DESCRIPTION_LABEL="Logout Description Text"
 JFIELD_BASIS_LOGOUT_DESCRIPTION_SHOW_DESC="Show or hide logout description."
 JFIELD_BASIS_LOGOUT_DESCRIPTION_SHOW_LABEL="Logout Description"
 JFIELD_CATEGORY_DESC="The category that this item is assigned to. You may select an existing category or enter a new category by typing the name in the field and pressing enter."
+JFIELD_COLOR_ERROR_CONVERT_HSL="Unable to convert HSL value"
+JFIELD_COLOR_ERROR_CONVERT_HUE="Unable to convert hue value"
+JFIELD_COLOR_ERROR_NO_COLOUR="No colour value available"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.ini]:
```suggestion
JFIELD_COLOR_ERROR_NO_COLOR="No colour value available"
```

**Review by nadjak77 — APPROVED** [review]

**Pull request merged**